wgpu-hal/src/gles/egl.rs

```diff
@@ -260,24 +264,41 @@ impl Inner {
         let (config, supports_native_window) = choose_config(&egl, display)?;
         egl.bind_api(egl::OPENGL_ES_API).unwrap();
 
+        let needs_robustness = true;
+        let mut khr_context_flags = 0;
+        let supports_khr_context = display_extensions.contains("EGL_KHR_create_context");
+
         //TODO: make it so `Device` == EGL Context
         let mut context_attributes = vec![
             egl::CONTEXT_CLIENT_VERSION,
             3, // Request GLES 3.0 or higher
         ];
-        // Debug requires EGL 1.5+
-        if flags.contains(crate::InstanceFlags::DEBUG) && version >= (1, 5) {
+        if flags.contains(crate::InstanceFlags::DEBUG) {
             log::info!("\tEGL context: +debug");
-            //TODO: figure out why this is needed
-            context_attributes.push(egl::CONTEXT_OPENGL_DEBUG);
-            context_attributes.push(egl::TRUE as _);
+            if version >= (1, 5) {
+                context_attributes.push(egl::CONTEXT_OPENGL_DEBUG);
+                context_attributes.push(egl::TRUE as _);
+            } else if supports_khr_context {
+                khr_context_flags |= EGL_CONTEXT_OPENGL_DEBUG_BIT_KHR;
+            }
         }
-        if display_extensions.contains("EGL_EXT_create_context_robustness") {
+        if needs_robustness {
             log::info!("\tEGL context: +robust access");
```

Comment:
could these debugs be moved into the if statements and carry some info about which version they're enabling, like for robustness they could say `+robust access`, `+robust access EXT`, and `+robust access KHR`.